### PR TITLE
Fixes a blacklisted job is forcefully removed from the game

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Fixes a blacklisted job is forcefully removed from the game
When a job shouldn't be used in a game, the current code removes it from the game
but we should let it exist anyway, because a job is used in DB or various systems
if a job shouldn't be used, the game just should not allow user to have it

Also, this includes some job code improvement.

## Why It's Good For The Game
bug fix

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

Only jobs you can play is available

image

image

CentCom can do that

</details>

## Changelog
:cl:
fix: Jobs will be told when a job is not available to play in a shift.
admin: CentCom console can manage all jobs now. You  can prioritize Captain role.
fix: fixed some job code glitches
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -30,6 +30,21 @@ SUBSYSTEM_DEF(job)
 	var/list/crew_obj_list = list()
 	var/list/crew_obj_jobs = list()
 
+	/// jobs that are not allowed in HoP job manager
+	var/list/manipulation_blacklist_jobs = list(
```
Comment (PowerfulBacon) [inline code/controllers/subsystem/job.dm]: This needs a better name, something like `locked_jobs` or `read_only_jobs`

File: tgui/packages/tgui/interfaces/PreferencesMenu/JobsPage.tsx [hunk 2]
```diff
@@ -134,10 +134,19 @@ const JobRow = (
...
-  if (experienceNeeded) {
+  if (isLocked) {
+    rightSide = (
+      <Stack align="center" height="100%" pr={1}>
+        <Stack.Item grow textAlign="right">
+          Locked by the server & map configuration
+        </Stack.Item>
+      </Stack>
+    );
+  } else if (experienceNeeded) {
```
Comment (PowerfulBacon) [inline tgui/packages/tgui/interfaces/PreferencesMenu/JobsPage.tsx]: Jobs that are disabled in the server config will never appear and shouldn't show at all. Only jobs disabled by the map should appear and it should say "Not available on this map" or something

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/job.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/controllers/subsystem/job.dm]: Those suggestions can misinform that the jobs are not available in the game due to "locked_flags" name existence, but these jobs are still usable.
I will pick `job_manager_blacklisted`

File: tgui/packages/tgui/interfaces/PreferencesMenu/JobsPage.tsx (hunk 2 above)
Comment (EvilDragonfiend) [reply tgui/packages/tgui/interfaces/PreferencesMenu/JobsPage.tsx]: image

**Review by PowerfulBacon — APPROVED** [review]

**Pull request merged**